go-controller/pkg/ovn/egressqos.go

```diff
@@ -864,6 +895,36 @@ func (oc *DefaultNetworkController) onEgressQoSNodeAdd(obj interface{}) {
 		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v: %v", obj, err))
 		return
 	}
+	node := obj.(*kapi.Node)
+	if util.GetNodeZone(node) != oc.zone {
+		return
+	}
+	oc.egressQoSNodeQueue.Add(key)
+}
+
+// onEgressQoSNodeUpdate queues the node for processing if it changed zones
+func (oc *DefaultNetworkController) onEgressQoSNodeUpdate(oldObj, newObj interface{}) {
+	oldNode := oldObj.(*kapi.Node)
+	newNode := newObj.(*kapi.Node)
+	if oldNode.ResourceVersion == newNode.ResourceVersion ||
+		!newNode.GetDeletionTimestamp().IsZero() {
+		return
+	}
+	// During a nodeAdd event, the ovnkube-node can take some time to add the zone
+	// annotation to the node, during that interim time we might consider the node
+	// as remote and hence the addNode event might not do anything. So we need to
+	// watch for node updates. We also ensure we only process local node zones by
+	// comparing to the controller's zone
+	oldNodeZone := util.GetNodeZone(oldNode)
+	newNodeZone := util.GetNodeZone(newNode)
+	if oldNodeZone == newNodeZone || newNodeZone != oc.zone {
```

Comment:
just making sure, a node moving from local to remote has the same effect as it being deleted and that's why we don't care about it here?

Reply:
correct. when a node moves from local to remote the nodeUpdate path will catch that and delete the switch, so there is nothing for us to do here.

Reply:
let me add a comment that makes this explicit.

Reply:
done.